hack/jenkins/e2e.sh

```diff
@@ -285,6 +285,7 @@ GKE_REQUIRED_SKIP_TESTS=(
     # Alpha features, remove from skip when these move to beta
     "Daemon\sset"
     "Deployment"
+    "ThirdParty"
```

Comment:
Yes, please change this instead to use `SkipIfProviderIs` or `SkipUnlessProviderIs`.